website/static/js/registrationUtils.js

```diff
@@ -15,51 +15,214 @@ var language = osfLanguage.registrations;
 
 var SaveManager = require('js/saveManager');
 var editorExtensions = require('js/registrationEditorExtensions');
-var registrationEmbargo = require('js/registrationEmbargo');
+var RegistrationModal = require('js/registrationModal');
+
+// This value should match website.settings.DRAFT_REGISTRATION_APPROVAL_PERIOD
+var DRAFT_REGISTRATION_MIN_EMBARGO_DAYS = 10;
+var DRAFT_REGISTRATION_MIN_EMBARGO_TIMESTAMP = new Date().getTime() + (
+        DRAFT_REGISTRATION_MIN_EMBARGO_DAYS * 24 * 60 * 60 * 1000
+);
+
+/**
+ * @class Comment
+ * Model for storing/editing/deleting comments on form fields
+ *
+ * @param {Object} data: optional data to instatiate model with
+ * @param {User} data.user
+ * @param {Date string} data.lastModified
+ * @param {Date string} data.created
+ * @param {Boolean} data.isDeleted
+ * @param {String} data.value
+ *
+ * @type User
+ * @property {String} id
+ * @property {String} fullname
+ **/
+function Comment(data) {
+    var self = this;
+
+    self.saved = ko.observable(data ? true : false);
+
+    data = data || {};
+    self.user = data.user || $osf.currentUser();
+    self.value = ko.observable(data.value || '');
+    self.value.subscribe(function() {
+        self.lastModified = new Date();
+    });
+
+    self.created = data.created ? new Date(data.created) : new Date();
+    self.lastModified = data.lastModified ? new Date(data.lastModified) : new Date();
+
+    self.isDeleted = ko.observable(data.isDeleted || false);
+    self.isDeleted.subscribe(function(isDeleted) {
+        if (isDeleted) {
+            self.value('');
+        }
+    });
+
+    self.seenBy = ko.observableArray([self.user.id]);
+
+    /**
+     * Returns true if the current user is the comment owner
+     **/
+    self.isOwner = ko.pureComputed(function() {
+        return self.user.id === $osf.currentUser().id;
+    });
+
+    /**
+     * Returns the author as the actual user, not 'You'
+     **/
+    self.author = ko.pureComputed(function() {
+        return self.user.fullname;
+    });
+
+    /**
+     * Returns 'You' if the current user is the commenter, else the commenter's name
+     */
+    self.getAuthor = ko.pureComputed(function() {
+        if (self.isOwner()) {
+            return 'You';
+        } else {
+            return self.user.fullname;
+        }
+    });
+
+    /**
+     * Returns true if the current user is the comment creator
+     **/
+    self.canDelete = ko.pureComputed(function() {
+        return self.isOwner();
+    });
+
+    /**
+     * Returns true if the comment is saved and the current user is the comment creator
+     **/
+    self.canEdit = ko.pureComputed(function() {
+        return !self.isDeleted() && self.saved() && self.isOwner();
+    });
+}
+/** Toggle the comment's save state **/
+Comment.prototype.toggleSaved = function(save) {
+    var self = this;
+
+    self.saved(!self.saved());
```

Comment:
If I'm reading this correctly, the comment is marked as saved *before* the operation has completed, and `save` appears to send an http request.

As a general rule, this flag should only toggle after successful completion of the operation.  If your save method returns a promise, this is an easy change- you could write:

```javascript
if (!self.saved()){
    result = save();
    result.done(self.saved(true));
};
```